dimod/constrained.py

```diff
@@ -618,6 +619,88 @@ def from_file(cls, fp: Union[BinaryIO, ByteString]) -> "ConstrainedQuadraticMode
 
         return cqm
 
+    def iter_violations(self, sample_like, *, skip_satisfied: bool = False, clip: bool = True,
+                        ) -> Iterator[Tuple[Hashable, Bias]]:
+        """Yield constraint labels and the amount the constraints are violated.
+
+        Args:
+            sample_like: A sample over the CQM variables.
```

Comment:
Why does this one specify "over the CQM variables"? Typically we'd just say "A raw sample. sample_like is an extension of NumPy’s array_like structure. See as_samples()."